### PR TITLE
Implement loading and saving of grouping definitions

### PR DESCRIPTION
**Pull request by bolotovskyr**
This PR implements two new `SNAPRed` algorithms - see [Story 1775](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1775):

- `SaveGroupingDefinition` for saving grouping workspaces in DiffCal (HDF5) format. 
- `LoadGroupingDefinition` for loading grouping definitions from NEXUS, XML, or DiffCal/HDF5 formats

All existent `SNAPRed` instances of loading grouping definitions from NEXUS or XML grouping files are now rerouted to `LoadGroupingDefinition`.

`PixelGroupingParametersCalculationAlgorithm` got new tests to exercise DiffCal/HDF5 formats. Most of this algorithm's tests are now run locally.

To exercise the new tests:

```
pytest -s tests/unit/backend/recipe/algorithm/test_LoadGroupingDefinition.py
pytest -s tests/unit/backend/recipe/algorithm/test_SaveGroupingDefinition.py
pytest -s tests/unit/backend/recipe/algorithm/test_PixelGroupingParametersCalculationAlgorithm.py
```

### CODE REVIEW

File: src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    def PyInit(self) -> None:
+        # declare properties
+        self.declareProperty(
+            "GroupingFilename",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="Path of an input grouping definition file",
+        )
+        self.declareProperty(
+            "InstrumentName", defaultValue="", direction=Direction.Input, doc="Name of an associated instrument"
```
Comment (rboston628) [inline src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py:31]: trailing comma at end

File: src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        )
+        self.declareProperty(
+            "InstrumentFilename",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="Path of an associated instrument definition file",
+        )
+        self.declareProperty(
+            MatrixWorkspaceProperty("InstrumentDonor", "", Direction.Input, PropertyMode.Optional),
```
Comment (rboston628) [inline src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py:40]: Does this need to be declared a `MatrixWorkspaceProperty`?  So far we've handled all workspace properties by their string names.

Comment (bolotovskyr) [reply src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py:40]: I copied that aspect from [this PR](https://github.com/neutrons/SNAPRed/pull/73/files#diff-d5a5b47544d5bb130b0bc0b6f79bc6e7717c08a51ae6d202e2af3bc036afa4b0R17). Pete has explained that adds more safety to the code as it verifies that the string name indeed represents a workspace.  Also, I don't see how else we could specify that a property is optional.

File: src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py (hunk 1 above)
Comment (rboston628) [reply src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py:31]: Change to 
``` python
self.declareProperty(
     "InstrumentName", 
    defaultValue="", 
    direction=Direction.Input, 
    doc="Name of an associated instrument",
)
```
Without the trailing comma, the linter will flatten it to one line.

Comment (bolotovskyr) [reply src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py]: Sure.  I was just waiting for more requests to bunch them all up in one commit :)

**Review by rboston628 — APPROVED** [review]

**Pull request merged**